src/servers/ZoneServer2016/entities/npc.ts

```diff
@@ -282,6 +283,10 @@ export class Npc extends BaseFullCharacter {
             this.triggerAwards(server, client, this.rewardItems);
             break;
         }
+        if (Math.random() <= 0.15) {
+          // 15% chance for knife to break upon harvest
+          server.removeInventoryItem(client.character, skinningKnife);
```

Comment:
maybe add a chattext about the skinningKnife being broken

Reply:
For sure, also to add-on about adding durability to the skinning knife - I implemented it and attempted to test it, but the way skinning knifes are treated makes it so they aren't a loadout item but any other generic BaseItem. So when you call the damageItem function, it doesn't actually do anything. This is because inside of damageItem, another function is called named updateLoadoutItem and switching around the logic to work with BaseItem still didn't work b/c skinning knifes are loadout items.

Should I still bother with the durability on skinning knifes? Or perhaps maybe there's another way to implement what I was attempting to?

Reply:
changes done to npc class (guess you had something similar):
```
OnPlayerSelect(server: ZoneServer2016, client: ZoneClient2016) {
    const skinningKnife = client.character.getItemById(Items.SKINNING_KNIFE);
    if (!this.isAlive && skinningKnife) {
      server.utilizeHudTimer(client, this.nameId, 5000, 0, () => {
        switch (this.actorModelId) {
          case ModelIds.ZOMBIE_FEMALE_WALKER:
          case ModelIds.ZOMBIE_MALE_WALKER:
            const emptySyringe = client.character.getItemById(
              Items.SYRINGE_EMPTY
            );
            if (emptySyringe) {
              client.character.lootContainerItem(
                server,
                server.generateItem(Items.SYRINGE_INFECTED_BLOOD)
              );
              server.removeInventoryItem(client.character, emptySyringe);
              return;
            }
            this.triggerAwards(server, client, this.rewardItems);
            break;
          case ModelIds.DEER_BUCK:
          case ModelIds.DEER:
            this.triggerAwards(server, client, this.rewardItems);
            break;
          case ModelIds.BEAR:
            this.triggerAwards(server, client, this.rewardItems);
            break;
          case ModelIds.WOLF:
            this.triggerAwards(server, client, this.rewardItems);
            break;
        }
        server.damageItem(client, skinningKnife, 400);
        server.deleteEntity(this.characterId, server._npcs);
      });
    }
  }
```
then you need to modify damageItem method in ZoneServer to work with base Items:

```
  damageItem(client: Client, item: LoadoutItem | BaseItem, damage: number) {
    if (item.itemDefinitionId == Items.WEAPON_FISTS) return;

    item.currentDurability -= damage;
    if (item.currentDurability <= 0) {
      this.removeInventoryItem(client.character, item);
      if (this.isWeapon(item.itemDefinitionId)) {
        client.character.lootContainerItem(
          this,
          this.generateItem(Items.BROKEN_METAL_ITEM)
        );
      }
      return;
    }
    if (item instanceof LoadoutItem) {
      this.updateLoadoutItem(client, item);
    } else if (item instanceof BaseItem) {
      const container = client.character.getItemContainer(item.itemGuid);
      if (!container) return;
      this.updateContainerItem(client.character, item, container);
    }
  }
```

finally generateItem in zoneServer, and pGetItemData in baseFullCharacter and Character classes need first switch modified to apply max durability:

```
case itemDefinitionId == Items.SKINNING_KNIFE:
        durability = 1000;
        break;
```

Reply:
Also we need to check at the end of utilizing hud timer if the client still has the item to avoid glitching (dropping to not loose a durability)

Reply:
> changes done to npc class (guess you had something similar):
> 
> ```
> OnPlayerSelect(server: ZoneServer2016, client: ZoneClient2016) {
>     const skinningKnife = client.character.getItemById(Items.SKINNING_KNIFE);
>     if (!this.isAlive && skinningKnife) {
>       server.utilizeHudTimer(client, this.nameId, 5000, 0, () => {
>         switch (this.actorModelId) {
>           case ModelIds.ZOMBIE_FEMALE_WALKER:
>           case ModelIds.ZOMBIE_MALE_WALKER:
>             const emptySyringe = client.character.getItemById(
>               Items.SYRINGE_EMPTY
>             );
>             if (emptySyringe) {
>               client.character.lootContainerItem(
>                 server,
>                 server.generateItem(Items.SYRINGE_INFECTED_BLOOD)
>               );
>               server.removeInventoryItem(client.character, emptySyringe);
>               return;
>             }
>             this.triggerAwards(server, client, this.rewardItems);
>             break;
>           case ModelIds.DEER_BUCK:
>           case ModelIds.DEER:
>             this.triggerAwards(server, client, this.rewardItems);
>             break;
>           case ModelIds.BEAR:
>             this.triggerAwards(server, client, this.rewardItems);
>             break;
>           case ModelIds.WOLF:
>             this.triggerAwards(server, client, this.rewardItems);
>             break;
>         }
>         server.damageItem(client, skinningKnife, 400);
>         server.deleteEntity(this.characterId, server._npcs);
>       });
>     }
>   }
> ```
> 
> then you need to modify damageItem method in ZoneServer to work with base Items:
> 
> ```
>   damageItem(client: Client, item: LoadoutItem | BaseItem, damage: number) {
>     if (item.itemDefinitionId == Items.WEAPON_FISTS) return;
> 
>     item.currentDurability -= damage;
>     if (item.currentDurability <= 0) {
>       this.removeInventoryItem(client.character, item);
>       if (this.isWeapon(item.itemDefinitionId)) {
>         client.character.lootContainerItem(
>           this,
>           this.generateItem(Items.BROKEN_METAL_ITEM)
>         );
>       }
>       return;
>     }
>     if (item instanceof LoadoutItem) {
>       this.updateLoadoutItem(client, item);
>     } else if (item instanceof BaseItem) {
>       const container = client.character.getItemContainer(item.itemGuid);
>       if (!container) return;
>       this.updateContainerItem(client.character, item, container);
>     }
>   }
> ```
> 
> finally generateItem in zoneServer, pGetItemData in baseFullCharacter and Character classes need first switch modified to apply max durability:
> 
> ```
> case itemDefinitionId == Items.SKINNING_KNIFE:
>         durability = 1000;
>         break;
> ```

Yess I had somewhat the same, I just wasn't aware of the updateContainerItem function 😅 And I changed the ITEM_TYPE of the skinning knife to 20 so it gets mixed in with the isWeapon function when applying durability. Idk if that would break anything but if so we could go for adding 👍 Thank you though and I'll look into the Hud Timer stuff when I get some time

Reply:
https://streamable.com/3uc5ss

I tested the glitch and fortunately damageItem still will damage the item even if a player tries to drop the item in the middle of the hudTimer, so the glitch wont happen :D 👍 

This PR should be good to go, just need to change the amount of durability loss on the skinning knife as 50 was supposed to be a placeholder. What do y'all reckon it should be? 100? 200? The skinning knife has durability of 2000 btw!

Reply:
> https://streamable.com/3uc5ss
> 
> I tested the glitch and fortunately damageItem still will damage the item even if a player tries to drop the item in the middle of the hudTimer, so the glitch wont happen :D 👍
> 
> This PR should be good to go, just need to change the amount of durability loss on the skinning knife as 50 was supposed to be a placeholder. What do y'all reckon it should be? 100? 200? The skinning knife has durability of 2000 btw!

idk for me it should be high but maybe 200 is too much so 100 ?